pandas/io/formats/style.py

```diff
@@ -802,6 +812,158 @@ def where(
+            )
+
+        ttips = ttips.reindex_like(self.data)
+
+        mask = (ttips.isna()) | (ttips.eq(""))
+        self.tooltip_styles = [
+            d
+            for sublist in [
+                [
+                    {
+                        "selector": "#T_"
+                        + self.uuid
+                        + "row"
+                        + str(i)
+                        + "_col"
+                        + str(j)
+                        + f":hover .{self.tooltip_class}",
+                        "props": [("visibility", "visible")],
+                    },
+                    {
+                        "selector": "#T_"
+                        + self.uuid
+                        + "row"
+                        + str(i)
+                        + "_col"
+                        + str(j)
+                        + f" .{self.tooltip_class}::after",
+                        "props": [("content", f'"{str(ttips.iloc[i, j])}"')],
+                    },
+                ]
+                for i, rn in enumerate(ttips.index)
+                for j, cn in enumerate(ttips.columns)
+                if not mask.iloc[i, j]
+            ]
+            for d in sublist
+        ]
+
+        return self
+
+    def set_tooltip_class(self, name="pd-t", properties=None) -> "Styler":
+        """
+        Method to set the name and properties of the class for creating tooltips on
+        hover.
+
+        Parameters
+        ----------
+        name : str, default 'pd-t'
+            Name of the tooltip class used in CSS, should conform to HTML standards.
+        properties : list-like, default None
+            List of (attr, value) tuples; see example. If `None` will use defaults.
+
+        Returns
+        -------
+        self : Styler
+
+        Notes
+        -----
+        Default properties for the tooltip class are as follows:
+
+        - visibility: hidden
+        - position: absolute
+        - z-index: 1
+        - background-color: black
+        - color: white
+        - transform: translate(-20px, -20px)
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(np.random.randn(10, 4))
+        >>> df.style.set_tooltip_class(name='tt-add', properties=[
+        ...     ('visibility', 'hidden'),
+        ...     ('position', 'absolute'),
+        ...     ('z-index', 1)])
+        """
+        if properties is None:
+            properties = [  # set default
+                ("visibility", "hidden"),
+                ("position", "absolute"),
+                ("z-index", 1),
+                ("background-color", "black"),
+                ("color", "white"),
+                ("transform", "translate(-20px, -20px)"),
+            ]
+        self.tooltip_class = name
+
+        self.tooltip_class_styles = [
+            {"selector": f".{self.tooltip_class}", "props": properties}
+        ]
+        return self
+
+    def _render_tooltips(self, d):
```

Comment:
does this make sense as a module level free function (where you pass in the styles & d)?